subsys/dfu/dfu_target/src/dfu_target_mcuboot.c

```diff
@@ -31,6 +31,8 @@ LOG_MODULE_REGISTER(dfu_target_mcuboot, CONFIG_DFU_TARGET_LOG_LEVEL);
 #define MCUBOOT_SECONDARY_LAST_PAGE_ADDR                                       \
 	(PM_MCUBOOT_SECONDARY_ADDRESS + PM_MCUBOOT_SECONDARY_SIZE - 1)
 
+#define IS_ALIGNED_32(POINTER) (((uintptr_t)(const void *)(POINTER)) % 4 == 0)
```

Comment:
Maybe just reuse the one from `zephyr/drivers/flash/soc_flash_nrf.c`

```suggestion
static inline bool is_alinged_32(uint32_t data)
{
        return (data & 0x3) ? false : true;
}

Reply:
That function is quite different and is checking a value vs a pointer. I tested the change I made to ensure it works when running on the hardware.
Warnings are produced if trying to use `is_alinged_32 `:
```
/Users/ryan/git/pinnacle100_workspace/nrf/subsys/dfu/dfu_target/src/dfu_target_mcuboot.c: In function 'dfu_target_mcuboot_set_buf':
/Users/ryan/git/pinnacle100_workspace/nrf/subsys/dfu/dfu_target/src/dfu_target_mcuboot.c:98:21: warning: passing argument 1 of 'is_aligned_32' makes integer from pointer without a cast [-Wint-conversion]
   98 |  if (!is_aligned_32(buf)) {
      |                     ^~~
      |                     |
      |                     uint8_t * {aka unsigned char *}
/Users/ryan/git/pinnacle100_workspace/nrf/subsys/dfu/dfu_target/src/dfu_target_mcuboot.c:86:43: note: expected 'uint32_t' {aka 'unsigned int'} but argument is of type 'uint8_t *' {aka 'unsigned char *'}
   86 | static inline bool is_aligned_32(uint32_t data)
```

Reply:
Since you are checking if the address of the pointer is aligned I don't see an issue with passing it as data by type casting it.
`if (!is_aligned_32((uint32_t)buf))` or if you prefer `if (!is_aligned_32((uintptr_t)buf))` as you did in your macro.

Functionally both the macro and function is doing the same doing a % 4 to check if it's 0 or not. 
However this is not a blocker for me so feel to keep it as is.

Reply:
Thanks, I will leave as is as I feel it is more clear.